WhisperAPI/WhisperAPI.Tests/Unit/NlpCallTest.cs

```diff
@@ -113,8 +114,81 @@ public void When_receive_ok_response_with_empty_content_from_post_then_returns_n
                     Content = new StringContent(JsonConvert.SerializeObject(string.Empty))
                 }));
 
-            var result = this._nlpCall.GetNlpAnalysis(sentence);
-            result.Should().BeEquivalentTo((NlpAnalysis)null);
+            var searchQuery = SearchQueryBuilder.Build.WithQuery(sentence).Instance;
+            this._nlpCall.UpdateAndAnalyseSearchQuery(searchQuery);
+        }
+
+        [Test]
+        [TestCase("*#")]
+        [TestCase("I*")]
+        [TestCase("*like*")]
+        [TestCase("I*i*C*")]
+        [TestCase("I like C#")]
+        public void When_Intent_With_and_without_Wildcard_is_irrelevant(string wildcardString)
+        {
+            var searchQuery = SearchQueryBuilder.Build.WithQuery("I like C#").Instance;
+            var intentsFromApi = new List<string>
+            {
+                wildcardString
+            };
+
+            var intentsFromNLP = new List<Intent>
+            {
+                IntentBuilder.Build.WithName("I like C#").Instance
+            };
+
+            var nlpAnalysis = NlpAnalysisBuilder.Build.WithIntents(intentsFromNLP).Instance;
+            const string baseAddress = "http://localhost:5000";
+
+            this._httpClient = new HttpClient(this._httpMessageHandlerMock.Object);
+            this._nlpCall = new NlpCall(this._httpClient, new List<string> { wildcardString }, baseAddress);
+
+            this._httpMessageHandlerMock.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .Returns(Task.FromResult(new HttpResponseMessage
+                {
+                    StatusCode = System.Net.HttpStatusCode.OK,
+                    Content = new StringContent(JsonConvert.SerializeObject(nlpAnalysis))
+                }));
+
+            this._nlpCall.UpdateAndAnalyseSearchQuery(searchQuery);
+            searchQuery.Relevant.Should().BeFalse();
+        }
+
+        [Test]
+        [TestCase("smalltalk.*")]
+        public void When_Intent_is_relevant(string wildcardString)
```

Comment:
Ce test au complet est inutile, on se sert pas de la wildcard. Ce que on devrait faire à la place, c'est copier le test plus haut, faire un test case qui ne contient pas "i like C#" et valider que  searchQuery.Relevant.Should().BeTrue();